File: P8.LargestProductInSeries/python/largest-product.py
#!/bin/python

import sys

def product(n):
    p = 1
    while n != 0:
        p *= n % 10
        if p == 0:
            return 0
        n /= 10
    return p


t = int(raw_input().strip())
for a0 in xrange(t):
    n,k = raw_input().strip().split(' ')
    n,k = [int(n),int(k)]
    num = raw_input().strip()
    largest = 0
    for i in xrange(n-k):
        p = product(long(num[i:i + k]))
        if (p > largest):
            largest = p
            
    print largest

File: P8.LargestProductInSeries/Java/LargestProduct.java
import java.io.*;
import java.util.*;
import java.text.*;
import java.math.*;
import java.util.regex.*;

public class LargestProduct {

    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        int t = in.nextInt();
        for(int a0 = 0; a0 < t; a0++){
            int n = in.nextInt();
            int k = in.nextInt();
            String num = in.next();
            
            long largest = 0;

            for (int i = 0; i < n - k; i++) {
                String p = num.substring(i, i + k);
                long product = product(Long.valueOf(p));
                if (product > largest)
                    largest = product;
            }

            System.out.println(largest);
        }
    }

    private static long product(long n){
        long p = 1;
        while(n != 0){
            p *= (n % 10);
            if (p == 0)
                return 0;
            n /= 10;
        }
        return p;
    }
}
